### PR TITLE
7 ticket api articles post comments

### PR DESCRIPTION
**Pull request by Anna702**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -174,3 +174,67 @@ describe("/api/articles/:article_id/comments", () => {
...
+
+  test("400: POST responds with right message when article does not exist", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "Interesting point of view, but I haven't read it",
+    };
+    return request(app)
+      .post("/api/articles/9999/comments")
+      .send(newComment)
+      .expect(400)
```
Comment (JVal-NC) [inline __tests__/app.test.js]: Would suggest this is a 404, the article id is valid but doesnt exist

File: __tests__/app.test.js [hunk 2]
```diff
@@ -174,3 +174,67 @@ describe("/api/articles/:article_id/comments", () => {
...
+
+  test("400: POST responds with right message when user does not exist", () => {
+    const newComment = {
+      username: "london_bridge",
+      body: "Interesting point of view, but I haven't read it",
+    };
+    return request(app)
+      .post("/api/articles/6/comments")
+      .send(newComment)
+      .expect(400)
```
Comment (JVal-NC) [inline __tests__/app.test.js]: Would suggest this is a 404, the user is valid but doesnt exist

File: __tests__/app.test.js [hunk 3]
```diff
@@ -174,3 +174,67 @@ describe("/api/articles/:article_id/comments", () => {
       });
   });
 });
+
+describe("POST: /api/articles/:article_id/comments", () => {
```
Comment (JVal-NC) [inline __tests__/app.test.js:178]: Good tests, would suggest a test for invalid id, ie a string.
Also a test for the post request to succeed if extra properties are provided on the request body, and are ignored.

**Review by JVal-NC — COMMENTED** [review]
A couple of extra tests to include but strong work so far

File: __tests__/app.test.js (hunk 3 above)
Comment (Anna702) [reply __tests__/app.test.js:178]: “a test for invalid id” - this scenario is already covered in GET / articles / article_id (via https://github.com/Anna702/NC-NEWS-project/blob/071c2f112fb3ecea6320b61087a8a983cf656578/controllers/errorHandlingControllers.js#L8 ) Do I need a new test regardless?

**Review by pippaamy — APPROVED** [review]
Good work Anna, lets merge it!

**Pull request merged**